database/rrdset.c

```diff
@@ -1895,6 +1862,30 @@ void rrdset_timed_done(RRDSET *st, struct timeval now) {
     netdata_thread_enable_cancelability();
 }
 
+time_t rrdset_set_update_every(RRDSET *st, time_t update_every) {
+
+    internal_error(true, "RRDSET '%s' switching update every from %d to %d",
+                   rrdset_id(st), (int)st->update_every, (int)update_every);
+
+    time_t prev_update_every = st->update_every;
```

Comment:
Hello guys, 

A small one from Coverity:

```sh
*** CID 361429:  High impact quality  (Y2K38_SAFETY)
/database/rrdset.c: 1871 in rrdset_set_update_every()
1865     time_t rrdset_set_update_every(RRDSET *st, time_t update_every) {
1866     
1867         internal_error(true, "RRDSET '%s' switching update every from %d to %d",
1868                        rrdset_id(st), (int)st->update_every, (int)update_every);
1869     
1870         time_t prev_update_every = st->update_every;
>>>     CID 361429:  High impact quality  (Y2K38_SAFETY)
>>>     A "time_t" value is stored in an integer with too few bits to accommodate it.  The expression "update_every" is cast to "int".
1871         st->update_every = update_every;
1872     
1873         // switch update every to the storage engine
1874         RRDDIM *rd;
1875         rrddim_foreach_read(rd, st) {
1876             for (size_t tier = 0; tier < storage_tiers; tier++) {
```

Reply:
`time_t` in 64 bit systems is 64 bits. Generally we need to work on this problem in a more generic way. For moment let's stick to `time_t` and we will fix it later.